storage/encoding/ssz.go

```diff
@@ -25,14 +25,10 @@ import (
 	"github.com/berachain/beacon-kit/primitives/constraints"
 	"github.com/berachain/beacon-kit/primitives/decoder"
 	"github.com/davecgh/go-spew/spew"
-	"github.com/karalabe/ssz"
 )
 
 // SSZValueCodec provides methods to encode and decode SSZ values.
-type SSZValueCodec[T interface {
-	ssz.StaticObject
```

Comment:
not strictly necessary but still simplfies a bit.
There is no reason why we could not encode `ssz.DynamicObject`s, so just kept `constraints.SSZMarshallable` that already contains `ssz.Object`